modules/internal/BytesStringCommon.chpl

```diff
@@ -183,6 +184,65 @@ module BytesStringCommon {
     return ret;
   }
 
+   /*
+   This function decodeHelp is used to create a wrapper for 
+   qio_decode_char_buf* and qio_decode_char_buf_esc and return 
+   the value of syserr , cp and nbytes.
+   asks for buffer , size of buffer , starting index to read buffer,
+   and choice between "qio_decode_char_buf" 
+   and "qio_decode_char_buf_esc" for escape statement
+   */
+
+   proc decodeHelp(buf:c_ptr(uint(8)), buflen:int, 
+                  bufstart:int, allowEsc: bool )
+   throws {
+    
+    //pragma "fn synchronization free"
+    extern proc qio_decode_char_buf(ref chr:int(32), 
+                                    ref nbytes:c_int,
+                                    buf:c_string,
+                                    buflen:ssize_t): syserr;
+    //pragma "fn synchronization free"
+    extern proc qio_decode_char_buf_esc(ref chr:int(32),
+                                        ref nbytes:c_int,
+                                        buf:c_string,
+                                        buflen:ssize_t): syserr;
+    // esc chooses between qio_decode_char_buf_esc and
+    // qio_decode_char_buf as a single wrapper function 
+    if(allowEsc){
+      var chr: int(32);
+      var nbytes: c_int;
+      var start = bufstart:c_int;
+      var multibytes = (buf + start+1): c_string;
+      var maxbytes = (buflen - start): ssize_t;
+      var decodeRet = qio_decode_char_buf_esc(chr, nbytes, 
+                                              multibytes,
+                                              maxbytes);
+       if decodeRet != 0 {  //decoder returns error
+          throw new owned DecodeError();
+        }
+       else{
+        return (chr, nbytes);
+      }
+    }
+    else{
+      var chr: int(32);
+      var nbytes: c_int;
+      var start = bufstart:c_int;
+      var multibytes = (buf + start+1): c_string;
+      var maxbytes = (buflen - start): ssize_t;
+      var decodeRet = qio_decode_char_buf(chr, nbytes,
+                                          multibytes, 
+                                          maxbytes);
+       if decodeRet != 0 {  //decoder returns error
+          throw new owned DecodeError();
+        }
+       else{
+        return (chr, nbytes);
+      }
+    }
```

Comment:
This if/else should be refactored so that the only thing in them is just calls to the extern proc. (I maybe missing something but they are the only different lines I think.)

Reply:
Let's make sure we address this